Documents WithWorkflowVersioningIntent (#2459)
temporalio/documentation: docs-src/go/how-to-use-worker-versioning-in-go.md

err := workflow.ExecuteActivity(ctx, YourActivityDefinition, yourActivityParam).Get(ctx, &yourActivityResult)
// ...
```

### Specifying Versioning for Continue-As-New

When using the Continue-As-New, specify the versioning intent:

> 1. Use the `WithWorkflowVersioningIntent` context modifier:

Reviewer: not sure it has to be a list anymore, but as long as the info is there :)

err := workflow.ExecuteActivity(ctx, YourActivityDefinition, yourActivityParam).Get(ctx, &yourActivityResult)
// ...
```

#### Specifying versions for Continue-As-New

> When using the Continue-As-New, use the `WithWorkflowVersioningIntent` context modifier.

Reviewer 2: ```suggestion
When using the Continue-As-New feature, use the `WithWorkflowVersioningIntent` context modifier.
```
Alternatively, "When using Continue-As-New, use...."

> The function `WithWorkflowVersioningIntent` sets the `VersioningIntentDefault` before constructing a `ContinueAsNewError` with `NewContinueAsNewError`:

Reviewer 2: ```suggestion
The function `WithWorkflowVersioningIntent` sets `VersioningIntentDefault` before constructing a `ContinueAsNewError` object with `NewContinueAsNewError`:
```
Or similar, per https://developers.google.com/style/code-in-text#grammatical-treatment-of-code-elements.

```go
ctx = workflow.WithWorkflowVersioningIntent(ctx, temporal.VersioningIntentDefault)
err := workflow.NewContinueAsNewError(ctx, "WorkflowName")
```

> If you're migrating Workflows between incompatible Worker Build IDs and you want your continued Workflows to start using the Task Queue's latest default version, utilize the `WithWorkflowVersioningIntent` as shown above before calling `NewContinueAsNewError`.

Reviewer 2: ```suggestion
If you're migrating Workflows between incompatible Worker Build IDs and you want your continued Workflows to start using the Task Queue's latest default version, use `WithWorkflowVersioningIntent` as shown earlier before calling `NewContinueAsNewError`.
```

err := workflow.ExecuteActivity(ctx, YourActivityDefinition, yourActivityParam).Get(ctx, &yourActivityResult)
// ...
```

### Specifying Versioning for Continue-As-New

When using the Continue-As-New, specify the versioning intent:

1. Use the `WithWorkflowVersioningIntent` context modifier:
   ```go

>    ctx = WithWorkflowVersioningIntent(ctx, temporal.VersioningIntentLatest)

Reviewer: ```suggestion
   ctx = workflow.WithWorkflowVersioningIntent(ctx, temporal.VersioningIntentDefault)
```
1. workflow code has to refer to the package name
2. it's called `VersioningIntentDefault`

   err := workflow.NewContinueAsNewError(ctx, "WorkflowName")
   ```

> 2. The function `WithWorkflowVersioningIntent` sets the `VersioningIntent` prior to constructing a `ContinueAsNewError` with `NewContinueAsNewError`:
>    ```go
>    func WithWorkflowVersioningIntent(ctx Context, intent temporal.VersioningIntent) Context {
>        return internal.WithWorkflowVersioningIntent(ctx, intent)
>    }
>    ```

Reviewer: there's no need to show how it's defined.. that doesn't add anything useful. maybe just use the text here on top of the first code block since it's a little more informative
```suggestion
```
